app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,27 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def metadata_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
+                                        and updates metadata with response_id
+        If response_id is present : return as it is
+
+        """
+        if data.get("response_id"):
+            return data
+        ru_ref = data.get("ru_ref")
+        collection_exercise_sid = data.get("collection_exercise_sid")
+        eq_id = data.get("eq_id")
+        form_type = data.get("form_type")
+        if ru_ref and collection_exercise_sid and eq_id and form_type:
```

Comment:
Would be half tempted to move the 2 optional ones first (ie. eq_id and form_type). Also could consider walrus operators, although the extra brackets make it look a big fugly

Reply:
moved optional up , don't think walrus operator is going to do much ere

Reply:
Could also just check you got `eq_id` and `form_type`, the other two are mandatory and must exist anyway.

Reply:
I agree with Walrus I removed the comment as the brackets made it look weird